google/cloud/aiplatform/prediction/local_endpoint.py

```diff
@@ -250,6 +251,109 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
+                container's environment. If a variable cannot be resolved, the reference
+                in the input string will be unchanged. The $(VAR_NAME) syntax can be
+                escaped with a double $$, ie: $$(VAR_NAME). Escaped references will
+                never be expanded, regardless of whether the variable exists or not.
+            serving_container_environment_variables (Dict[str, str]):
+                Optional. The environment variables that are to be present in the container.
+                Should be a dictionary where keys are environment variable names
+                and values are environment variable values for those names.
+            serving_container_ports (Sequence[int]):
+                Optional. Declaration of ports that are exposed by the container. This field is
+                primarily informational, it gives Vertex AI information about the
+                network connections the container uses. Listing or not a port here has
+                no impact on whether the port is actually exposed, any port listening on
+                the default "0.0.0.0" address inside a container will be accessible from
+                the network.
+            credential_path (str):
+                Optional. The path to the credential key that will be mounted to the container.
+                If it's unset, the environment variable, GOOGLE_APPLICATION_CREDENTIALS, will
+                be used if set.
+            host_port (str):
+                Optional. The port on the host that the port, AIP_HTTP_PORT, inside the container
+                will be exposed as. If it's unset, a random host port will be assigned.
+            container_ready_timeout (int):
+                Optional. The timeout in second used for starting the container or succeeding the
+                first health check.
+            container_ready_check_interval (int):
+                Optional. The time interval in second to check if the container is ready or the
+                first health check succeeds.
+        """
+        local_endpoint = LocalEndpoint(
+            serving_container_image_uri=serving_container_image_uri,
+            artifact_uri=artifact_uri,
+            serving_container_predict_route=serving_container_predict_route,
+            serving_container_health_route=serving_container_health_route,
+            serving_container_command=serving_container_command,
+            serving_container_args=serving_container_args,
+            serving_container_environment_variables=serving_container_environment_variables,
+            serving_container_ports=serving_container_ports,
+            credential_path=credential_path,
+            host_port=host_port,
+            container_ready_timeout=container_ready_timeout,
+            container_ready_check_interval=container_ready_check_interval,
+        )
+        return local_endpoint.__enter__()
+
+    def undeploy(self):
```

Comment:
I'd name this "stop()". Also, `__exit__` should call this, not the other way around.